pyomo/contrib/community_detection/detection.py

```diff
@@ -168,7 +170,225 @@ def detect_communities(model, type_of_community_map='constraint', with_objective
+                pos = nx.spring_layout(model_graph)
+
+        # Define color_map
+        color_map = cm.get_cmap('viridis', len(numbered_community_map))
+
+        # Create the figure and draw the graph
+        fig = plt.figure()
+        nx.draw_networkx_nodes(model_graph, pos, nodelist=node_list, node_size=40, cmap=color_map, node_color=color_list)
+        nx.draw_networkx_edges(model_graph, pos, alpha=0.5)
+
+        # Make the main title
+        graph_type = type_of_graph.capitalize()
+        community_map_type = self.type_of_community_map.capitalize()
+        main_graph_title = "%s graph - colored using %s community map" % (graph_type, community_map_type)
+
+        main_font_size = 14
+        plt.suptitle(main_graph_title, fontsize=main_font_size)
+
+        # Define a dict that will be used for the graph subtitle
+        subtitle_naming_dict = {'bipartite': 'Nodes are variables and constraints & Edges are variables in a constraint',
+                                'constraint': 'Nodes are constraints & Edges are common variables',
+                                'variable': 'Nodes are variables & Edges are shared constraints'}
+
+        # Make the subtitle
+        subtitle_font_size = 11
+        plt.title(subtitle_naming_dict[type_of_graph], fontsize=subtitle_font_size)
+
+        # Return the figure and pos, the position dictionary used for the graph layout
+        return fig, pos
+
+    def stringify(self):
+        """
+        This function takes in a community map of Pyomo components and returns the same community map but with the strings
+        of the Pyomo components. Alternatively, if the user does not provide a community map, this function can take in
+        a model and then create a community map (using the function detect_communities). Then, the same process of
+        "stringifying" the community map is performed, again resulting in a community map but with the strings
+        of the Pyomo components.
+
+        Returns
+        -------
+        str_community_map: dict
+            a Python dictionary that maps arbitrary keys (in this case, integers from zero to the number of
+            communities minus one) to two-list tuples containing the strings of the Pyomo components in the given community
+        """
+
+        # Create str_community_map and give it values that are the strings of the components in community_map
+        str_community_map = dict()
+        for key in self.community_map:
+            str_community_map[key] = ([str(component) for component in self.community_map[key][0]],
+                                      [str(component) for component in self.community_map[key][1]])
+
+        # Return str_community_map, which is identical to community_map except it has the strings of all of the Pyomo
+        # components instead of the actual components
+        return str_community_map
 
 
 def visualize_model_graph(model, community_map=None, type_of_graph='constraint', with_objective=True,
```

Comment:
Again, if the community map includes some of the parameter information already, why input it again here?